tools/jbrowse2/jbrowse2.py

```diff
@@ -0,0 +1,1619 @@
+                            % ddl
+                        )
+                if drdict.get("refName", None):
+                    # TODO displayedRegions is not just zooming to the region, it hides the rest of the chromosome
+                    view_json["displayedRegions"] = [
+                        drdict,
+                    ]
+                    logging.info("@@@ defaultlocation %s for default session" % drdict)
+                else:
+                    logging.info(
+                        "@@@ no contig name found for default session - please add one!"
+                    )
+                session_views.append(view_json)
+        session_name = default_data.get("session_name", "New session")
+        for key, value in mapped_chars.items():
+            session_name = session_name.replace(value, key)
+        session_json["name"] = session_name
+
+        if "views" not in session_json:
+            session_json["views"] = session_views
+        else:
+            session_json["views"] += session_views
+
+        config_json["defaultSession"] = session_json
+        self.config_json.update(config_json)
+
+        with open(self.config_json_file, "w") as config_file:
+            json.dump(self.config_json, config_file, indent=2)
+
+    def add_general_configuration(self, data):
+        """
+        Add some general configuration to the config.json file
+        """
+
+        config_path = self.config_json_file
+        if os.path.exists(config_path):
+            with open(config_path, "r") as config_file:
+                config_json = json.load(config_file)
+        else:
+            config_json = {}
+        if self.config_json:
+            config_json.update(self.config_json)
+        config_data = {}
+
+        config_data["disableAnalytics"] = data.get("analytics", "false") == "true"
+
+        config_data["theme"] = {
+            "palette": {
+                "primary": {"main": data.get("primary_color", "#0D233F")},
+                "secondary": {"main": data.get("secondary_color", "#721E63")},
+                "tertiary": {"main": data.get("tertiary_color", "#135560")},
+                "quaternary": {"main": data.get("quaternary_color", "#FFB11D")},
+            },
+            "typography": {"fontSize": int(data.get("font_size", 10))},
+        }
+        if not config_json.get("configuration", None):
+            config_json["configuration"] = {}
+        config_json["configuration"].update(config_data)
+        self.config_json.update(config_json)
+        with open(config_path, "w") as config_file:
+            json.dump(self.config_json, config_file, indent=2)
+
+    def clone_jbrowse(self, realclone=True):
+        """Clone a JBrowse directory into a destination directory. This also works in Biocontainer testing now
+        Leave as True between version updates on temporary tools - requires manual conda trigger :(
```

Comment:
is this comment still valid? `JB2VER` is v2.10.3 but the Galaxy tool version is v2.10.1 ...

Reply:
This needs internet? Should we switch to the offline version?

Reply:
Offline is better - however, was using the pinned version _to avoid changing the JB2 dependency_ while it was a temporary EU tool because of the fuss with new conda deps. 

So yes - should be 2.10.3 version of the depenency and no clone IMHO - thanks.